docs/architecture/engine.md

```diff
@@ -0,0 +1,197 @@
+# Storage Engine Overview
+
+M3DB is a time series database that was primarily designed to be horizontally scalable and handle a large volume of monitoring time series data.
+
+## Caveats / Limitations
```

Comment:
Could you move this section to the bottom of this document, or better yet, break it into its own doc